src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java

```diff
@@ -1057,6 +1060,31 @@ public Object unmarshal(HierarchicalStreamReader reader, UnmarshallingContext co
         }
     }
 
+    /** Custom converter for the special metadata map. */
+    class VendorParameters extends BreifMapConverter {
```

Comment:
```suggestion
    class VendorParametersConverter extends BreifMapConverter {
```

But I'm wondering if you couldn't just use BreifMapConverter directly. Having the vendor parameters as null will be a common occurrence in all pre-existing data directories, I'd rather deal with that elsewhere (e.g., in the caller code, or in the getter for that field).

Reply:
I tend to belong to the school of thought that nulls are evil otherwise you never know if a field is null because it is not set or a program error caused it to be that way. 

I will test to see if I can change the getter to return an empty map remove this class.